pyproject.toml

```
@@ -24,7 +24,7 @@ dependencies = [
   'pooch',
   'scooby>=0.5.1',
   'typing-extensions',
-  'vtk',               # keep without version constraints
+  'vtk<9.4.0',
```

Based on https://github.com/pyvista/pyvista/pull/6872#issuecomment-2501547036, maybe it's better to do 
```suggestion
  'vtk<9.3.2',
```

---

I think restricting the minor version is best for now. Limiting the patch release feels a little too restrictive to me, but I'm open to do this.

In the meantime, I vote we release this limit on 9.4 and come back to this

---

Since the current cycle is a minor release it is moot for the near term use, the next update to this line will either be `<9.4.1` or `<9.5.0`.  We will have to decide at that point.  If vtk has breaking changes, Id also lean towards pinning down to the patch level.

---

While VTK regularly issues breaking changes and regressions across minor releases, I feel this doesn't happen so often during patch releases (though I don't have the data on hand to support this feeling)

---

Maybe keep vtk in `pinned` then and do `'vtk<9.3.2'` there? I.e. adopt a convention of pinning vtk to minor releases generally, but pin vtk to patch releases for testing. Something to consider.

---

If we are unsure,  then I'd vote to pin minor versions only.  If major breaking changes are found in patch releases, then we pin patch releases going forward too.